### ISSUE TITLE
Support IP Address binding

### ISSUE THREAD

**Issue by GoogleCodeExporter** [sagarswathi/h2database#10]
```
What steps will reproduce the problem?
N/A

What is the expected output? What do you see instead?
To allow machine with multiple IP addresses to start H2 database server
with the same port in different addresses.

Add startup parameter, e.g. -webAddress, -tcpAddress etc.

What version of the product are you using? On what operating system, file
system, and virtual machine?
N/A

Do you know a workaround?
Use different port for different instances

How important/urgent is the problem for you?
Not very urgent, but when I deploy embedded H2 databases in an application
to two JBoss instance on the same machine (binding to two different IP
address), I have to change port number for two deployment, which is not
convenient. So, it's quite urgent. And the change is trivial anyway.

In your view, is this a defect or a feature request?
Feature request

Please provide any additional information below.
My initial email to the group:
http://groups.google.com/group/h2-database/browse_thread/thread/5c5a9d0a87c31c5f

JBoss configuration document
http://wiki.jboss.org/wiki/Wiki.jsp?page=ConfiguringMultipleJBossInstancesOnOneM
achine
remarks: it uses a -b <ip-addressORhost> parameter

Sample code in JBoss on creating Socket
http://viewvc.jboss.org/cgi-bin/viewvc.cgi/jbossas/trunk/jnp/src/main/org/jnp/in
terfaces/TimedSocketFactory.java?revision=7609&view=markup&pathrev=35186

Socket JavaDoc
http://java.sun.com/j2se/1.4.2/docs/api/java/net/Socket.html

The H2 Code that need to be changed:
http://h2database.googlecode.com/svn/trunk/h2/src/main/org/h2/util/NetUtils.java
instead of : new Socket(address, port);
uses:   Socket(String host, int port, InetAddress localAddr, int localPort) 

And take a parameter.
```

Original issue reported on code.google.com by `mingfai...@gmail.com` on 8 Jan 2008 at 4:40

**Comment by GoogleCodeExporter** [comment]
```
This has been implemented in the trunk (it will be included in the next 
release). The
bind IP address can now be set when using multi-homed host (if multiple network
adapters are available) using the system property h2.bindAddress.
```

Original comment by `thomas.t...@gmail.com` on 10 Jan 2008 at 9:23
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
Original comment by `thomas.t...@gmail.com` on 10 Jan 2008 at 9:23
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
There is a problem with the h2.bindAddress build. When it is binding to 
127.0.0.1,
the bind function works perfectly and I could confirm it no longer listen to the
ports of other IP addresses. However, when it is binding to normal IP 
addresses, it
always got the following exception. 

Caused by: org.h2.jdbc.JdbcSQLException: Connection is broken [90067-64]
    at org.h2.message.Message.getSQLException(Message.java:89)

Remarks: I build with JDK 1.6



```

Original comment by `mingfai...@gmail.com` on 13 Jan 2008 at 10:48

**Comment by GoogleCodeExporter** [comment]
```
Thanks for your help! I found and fixed the problem in the current trunk. 
Binding
actually worked, but there was some 'verify if the connection is up' code that 
tried
to connect to the wrong address... This is now fixed. Also, the IP addresses of 
the
servers are now listed correctly when starting from the command line.
```

Original comment by `thomas.t...@gmail.com` on 16 Jan 2008 at 10:29

**Comment by GoogleCodeExporter** [comment]
```
This has been implemented in version 1.0.65 (2008-01-18)
```

Original comment by `thomas.t...@gmail.com` on 18 Jan 2008 at 6:09
- Changed state: **Fixed**
